### ISSUE TITLE
Update any code that uses fcp:selectMenu() to use fcp:doSelectMenu() instead

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#1991]
This works in `fcp:doSelectMenu()` but not `fcp:selectMenu()`:

<img width="936" alt="Screen Shot 2019-09-24 at 8 39 21 am" src="https://user-images.githubusercontent.com/22286696/65468352-27ca8800-dea7-11e9-8187-d46e39994b19.png">

**Comment by latenitefilms** [comment]
@randomeizer - I wonder if there's a way we could make Rx statement, optionally synchronous? 

For example, something like `require("cp.apple.finalcutpro"):doSelectMenu({"Window", "Workspaces", "Test Workspace"}):Synchronous()` would wrap the Rx code in a synchronous function, so that we can use that in a `fcp:selectMenu()` wrapper? That way we can potentially avoid a lot of double-up code?

**Comment by randomeizer** [comment]
Yeah, can't be done really. The whole point is that they are quite likely run on a timer. Once that's done, you have to exit the current process to actually run the enxt one.

It's actually possible for an individual Rx task to be synchronous, but there is no guarantee that will be the case for any given statement.

**Comment by randomeizer** [comment]
The other thing is that some of the issues we've had with menus in the past is because we try to access them synchronously. Some menus only update their value after it's been executed and FCPX has gone and done something else.

Basically, we need to port our existing code over to Rx, which would be easier if we get error reporting to improve the debug messages.

**Comment by latenitefilms** [comment]
Couldn't you just do something like:

```lua
local hasFinished = false
fcp:doSelectMenu({"Window", "Workspaces", "Test Workspace"}):Then(function()
hasFinished = true
end):Now()
just.waitUntil(function() return hasFinished end)
```

...but yes, I agree, it would be better if we could just remove `fcp:selectMenu()` all together. We just still use it a LOT currently.

**Comment by randomeizer** [comment]
Ugh, maybe? I'm not sure if it would actually let the 'doSelectMenu' run, or if the 'waitUntil' would hog the thread. Seems like the worst possible way to run that kind of code though. Even if it works, it will be a CPU sucker.

**Comment by latenitefilms** [comment]
Fair enough - let's aim to just replace any `fcp:selectMenu()` code with `fcp:doSelectMenu()`. It's used quite a bit, so this issue might take a while to action.
